pyathena/sqlalchemy_athena.py

```diff
@@ -398,69 +498,53 @@ def get_view_names(self, connection, schema=None, **kw):
         except NoSuchTableError:
             return False
 
-    @reflection.cache
     def get_columns(self, connection, table_name, schema=None, **kw):
-        raw_connection = self._raw_connection(connection)
-        schema = schema if schema else raw_connection.schema_name
-        query = """
-                SELECT
-                  table_schema,
-                  table_name,
-                  column_name,
-                  data_type,
-                  is_nullable,
-                  column_default,
-                  ordinal_position,
-                  comment
-                FROM information_schema.columns
-                WHERE table_schema = '{schema}'
-                AND table_name = '{table}'
-                """.format(
-            schema=schema, table=table_name
-        )
-        retry_config = raw_connection.retry_config
-        retry = tenacity.Retrying(
-            retry=retry_if_exception(
-                lambda exc: self._retry_if_data_catalog_exception(  # type: ignore
-                    exc, schema, table_name
-                )
-            ),
-            stop=stop_after_attempt(retry_config.attempt),
-            wait=wait_exponential(
-                multiplier=retry_config.multiplier,
-                max=retry_config.max_delay,
-                exp_base=retry_config.exponential_base,
-            ),
-            reraise=True,
-        )
-        try:
-            return [
-                {
-                    "name": row.column_name,
-                    "type": _TYPE_MAPPINGS.get(
-                        self._get_column_type(row.data_type), NULLTYPE
-                    ),
-                    "nullable": True if row.is_nullable == "YES" else False,
-                    "default": row.column_default
-                    if not self._is_nan(row.column_default)
-                    else None,
-                    "ordinal_position": row.ordinal_position,
-                    "comment": row.comment,
-                }
-                for row in retry(connection.execute, query).fetchall()
-            ]
-        except OperationalError as e:
-            if not self._retry_if_data_catalog_exception(e, schema, table_name):
-                raise NoSuchTableError(table_name) from e
-            else:
-                raise e
+        # TODO: rewrite using new meta data API
+        metadata = self._get_table(connection, table_name, schema=schema, **kw)
+        columns_specs = []
+        for column in metadata.columns:
+            columns_specs.append({
+                "name": column.name,
+                "type": _TYPE_MAPPINGS.get(
+                    self._get_column_type(column.type), NULLTYPE
+                ),
+                "nullable": True,  # True if row.is_nullable == "YES" else False,
+                "default": None,
+                # "default": row.column_default
+                # if not self._is_nan(row.column_default)
+                # else None,
```

Comment:
Removed `'ordinal_position'`, not part of [the specification](https://docs.sqlalchemy.org/en/14/core/reflection.html?highlight=get_columns#sqlalchemy.engine.reflection.Inspector.get_columns) (for neither of SQLAlchemy 1.3, 1.4 or 2.0). Also given that Athena does not support any of `default` or `nullable`, lets not pretend it does.

- [ ] Clean-up

```suggestion
                "nullable": True,
                "default": None,
```